src/netaddress.cpp

```diff
@@ -322,13 +330,7 @@ enum Network CNetAddr::GetNetwork() const
     if (!IsRoutable())
         return NET_UNROUTABLE;
 
-    if (IsIPv4())
```

Comment:
Good question - no. But it was not necessary before this PR either. Some good simplification can be done here even on `master`. I will see whether it is better to add a new commit to this PR or file it as a separate PR.

Reply:
Correction - it is needed, otherwise this test would fail:

https://github.com/bitcoin/bitcoin/blob/476436b/src/test/netbase_tests.cpp#L45

Anyway, I made some simplifications on top of this PR that remove the "extending" of `enum Network` with `NET_UNKNOWN` and `NET_TEREDO` and also removed `GetExtNetwork()`. To avoid distractions, I will not append those commits to this PR but would rather open a new PR against `master`, once this PR gets merged.